src/Microsoft.AspNet.Mvc.Razor/RazorViewEngine.cs

```diff
@@ -361,5 +392,16 @@ private static bool IsApplicationRelativePath(string name)
             Debug.Assert(!string.IsNullOrEmpty(name));
             return name[0] == '~' || name[0] == '/';
         }
+
+        private static bool IsRelativePath(string name)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(name));
+
+            // **Reviewers**: Though ./ViewName looks like a relative path, checking for '/' leads to problems with
```

Comment:
Do we need a better data type to represents paths here so that things like view components can be more explicit that the path is relative?